src/iop/filmicrgb.c

```diff
@@ -128,19 +129,19 @@ typedef struct dt_iop_filmic_rgb_spline_t
 
 typedef struct dt_iop_filmicrgb_params_t
 {
-  float grey_point_source;
-  float black_point_source;
-  float white_point_source;
-  float security_factor;
-  float grey_point_target;
-  float black_point_target;
-  float white_point_target;
-  float output_power;
-  float latitude;
-  float contrast;
-  float saturation;
-  float balance;
-  int preserve_color;
+  float grey_point_source;  // $MIN: 0 $MAX: 100 $DEFAULT: 9.225 $DESCRIPTION: "middle grey luminance"
+  float black_point_source; // $MIN: -16 $MAX: -0.1 $DEFAULT: -10.55 $DESCRIPTION: "black relative exposure"
+  float white_point_source; // $MIN: 0 $MAX: 16 $DEFAULT: 3.45 $DESCRIPTION: "white relative exposure"
+  float security_factor;    // $MIN: -50 $MAX: 200 $DEFAULT: 0 $DESCRIPTION: "dynamic range scaling"
+  float grey_point_target;  // $MIN: .1 $MAX: 50 $DEFAULT: 18.45 $DESCRIPTION: "target middle grey"
+  float black_point_target; // $MIN: 0 $MAX: 100 $DEFAULT: 0 $DESCRIPTION: "target black luminance"
+  float white_point_target; // $MIN: 0 $MAX: 100 $DEFAULT: 100 $DESCRIPTION: "target white luminance"
+  float output_power;       // $MIN: 1 $MAX: 10 $DEFAULT: 5.98 $DESCRIPTION: "target power transfer function"
+  float latitude;           // $MIN: 0.01 $MAX: 100 $DEFAULT: 45
+  float contrast;           // $MIN: 0 $MAX: 5 $DEFAULT: 1.30
+  float saturation;         // $MIN: -50 $MAX: 200 $DEFAULT: 5 $DESCRIPTION: "extreme luminance saturation"
+  float balance;            // $MIN: -50 $MAX: 50 $DEFAULT: 12 $DESCRIPTION: "shadows/highlights balance"
+  int preserve_color;       // $DESCRIPTION: "preserve chrominance" $DEFAULT: 3
```

Comment:
it would be nice if we could use the `enum` type mnemonic instead of the number in there.

Reply:
Well, in this particular case, preserve_color isn't even an enum. If it were, then the combobox will automatically be filled with the enum values (houz is working on making human readable text from $DESCRIPTION comments available). Not sure how easy using mnemonics in the $DEFAULT comment would be in that case; this is perl code, so I'm doing houz the favor of not touching it :-)

Reply:
Thank you. ;-)
I just pushed fb46a038203f69d4509c360d32cc0de73773a1c4 which adds (among many other improvements) descriptions to the enum field. You can also just use enum mnemonics in `$DEFAULT` as the value is just copied over verbatim to the C/C++ file. You can have a look at the generated files `build/src/iop/introspection_*` while playing with those magic comments in the iop, maybe that makes it more clear.